docs/provider_integrations.md

```diff
@@ -0,0 +1,266 @@
+# Provider integrations
+
+Ding2 supports integration with other business systems for handling different aspects of a library website such as searching the library catalogue, determining the status of titles and authenticating patrons. Within Ding2 a system handling one or more of these aspects is called a *provider system*.
+
+Provider systems are integrated with Drupal by implementing one or more Drupal modules which use the `ding_provider` module. Such a module is called a *provider module*.
+
+Drupal modules requiring data or functionality from a business system must declare this requirement by implementing a `ding_provider_user` hook. This requirement must be met by other Drupal modules which integrate with business system and expose their data and functionality. This is declared by implementing a `ding_provider` hook.
+
+## Example of a provider integration
+
+The `ding_reservation` module is responsible for allowing users to manage their reservations of titles on the library website. The module exposed a user interface but does not store any reservations on its own. It requires a business system to handle that.
+
+Consequently the module declares this requirement by implementing a hook:
+
+```php
+/**
+ * Implements hook_ding_provider_user().
+ */
+function ding_reservation_ding_provider_user() {
+  return array(
+    'reservation' => array(
+      'required' => TRUE,
+      'install time setup' => TRUE,
+    ),
+  );
+}
+```
```

Comment:
Mayby a comment about the parameters documentation for the two hooks can be found in https://github.com/ding2/ding2/blob/master/modules/ding_provider/ding_provider.api.php

Reply:
:+1: - see b96ec80a7b69c4cffb831f3c7cf4e21006e3238f.